xformers/ops/fmha/common.py

```diff
@@ -192,11 +192,13 @@ def validate_inputs(self) -> None:
                 and self.value.shape == (B, Mkv, Kv)
             )
         H = self.query.shape[-2]
+        Hkv = self.key.shape[-2]
         if self.query.ndim == 4:  # BMHK
             valid_shapes = (
                 self.query.shape == (B, Mq, H, K)
-                and self.key.shape == (B, Mkv, H, key_embed_dim)
-                and self.value.shape == (B, Mkv, H, Kv)
+                and self.key.shape == (B, Mkv, Hkv, key_embed_dim)
+                and self.value.shape == (B, Mkv, Hkv, Kv)
+                and H % Hkv == 0
             )
```

Comment:
This is to support GQA right? The way we support GQA with xFormers in `memory_efficient_attention` is through 5-D inputs